libs/backend/src/cast_vote_records/export.ts

```diff
@@ -423,6 +429,76 @@ async function exportSignatureFileToUsbDrive(
   return ok();
 }
 
+/**
+ * Updates the creation timestamp of a directory and its children files (assuming no
+ * sub-directories) using the one method guaranteed to work:
+ * ```
+ * cp -r <directory-path> <directory-path>-temp
+ * rm <directory-path>
+ * mv <directory-path>-temp <directory-path>
+ * ```
+ */
+async function updateCreationTimestampOfDirectoryAndChildrenFiles(
+  directoryPath: string
+): Promise<void> {
+  const tempDirectoryPath = `${directoryPath}-temp`;
+  await fs.mkdir(tempDirectoryPath);
+  const fileNames = (await fs.readdir(directoryPath, { withFileTypes: true }))
+    .filter((entry) => entry.isFile())
+    .map((entry) => entry.name);
+  for (const fileName of fileNames) {
+    await fs.copyFile(
+      path.join(directoryPath, fileName),
+      path.join(tempDirectoryPath, fileName)
+    );
+  }
+  await fs.rm(directoryPath, { recursive: true });
+  await fs.rename(tempDirectoryPath, directoryPath);
+}
+
+/**
+ * File creation timestamps could reveal the order in which ballots were cast. To maintain
+ * voter privacy, every time a ballot is cast, we randomly select 1 or 2 existing cast vote records
+ * and update their creation timestamps to the present. This approach is equivalent to taking
+ * random ballots from a stack and placing them on top every time a new ballot is added to the
+ * stack, a kind of shuffling as we go.
+ *
+ * This shuffling as we go is irrelevant for full exports, where we already iterate over cast vote
+ * records in an order independent of creation timestamp.
+ */
+async function maintainVoterPrivacy(
+  exportContext: ExportContext
+): Promise<void> {
+  const { usbMountPoint } = exportContext;
+
+  const exportDirectoryPath = path.join(
+    usbMountPoint,
+    getExportDirectoryPathRelativeToUsbMountPoint(exportContext)
+  );
+  const castVoteRecordIds = (
+    await fs.readdir(exportDirectoryPath, { withFileTypes: true })
+  )
+    .filter((entry) => entry.isDirectory())
+    .map((entry) => entry.name);
+
+  if (castVoteRecordIds.length < 2) {
+    return;
+  }
+
+  const oneOrTwo = Math.random() < 0.5 ? 1 : 2;
+  for (let i = 0; i < oneOrTwo; i += 1) {
+    const randomCastVoteRecordDirectoryPath = path.join(
+      exportDirectoryPath,
+      assertDefined(
+        castVoteRecordIds[Math.floor(Math.random() * castVoteRecordIds.length)]
```

Comment:
Another `Math.random()`. Maybe add a function like this to use in both places?

```ts
function getRandomNumber({ min, max }: { min: number; max: number }): number {
  if (min >= max) {
    throw new Error('min must be less than max');
  }

  const range = max - min;
  if (range <= 0) {
    throw new Error('Range must be greater than 0');
  }

  const byteArray = randomBytes(4); // 4 bytes for a 32-bit integer
  const randomInt = byteArray.readUInt32LE(0); // Convert the bytes to an integer
  return min + (randomInt % range);
}
```

Then you can do this:
```suggestion
        castVoteRecordIds[getRandomNumber({ min: 0, max: castVoteRecordIds.length - 1 })]
```

Reply:
Good call - added a helper to libs/basics: https://github.com/votingworks/vxsuite/pull/3972/commits/1c59eb2dcd23e8db25b5c0fbd11d89eec642004e